Improve documentation (#115)
fastify/csrf-protection: README.md

@fastify/csrf-protection provides a series of utilities that developers can use to secure their application.
We recommend using [@fastify/helmet](https://github.com/fastify/fastify-helmet) to implement some of those mitigations.

> Security is always a tradeoff between risk mitigation, functionality, performance and developer experience.

Reviewer: ```suggestion
Security is always a tradeoff between risk mitigation, functionality, performance, and developer experience.
```

### `fastify.csrfProtection(request, reply, next)`

A hook that you can use for protecting routes or entire plugins from CSRF attacks.

> Generally, we recommend to use a `onRequest` hook, but if you are sending the token

Reviewer: ```suggestion
Generally, we recommend using an `onRequest` hook, but if you are sending the token
```

}
```

> It is recommended to provide a custom `getToken` function for security and performance reasons.

Reviewer 2: I would link some OWASP documentation like 

https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers

Author: Like this?

```suggestion
It is recommended to provide a custom `getToken` function for performancs and [security](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers) reasons.
```

Author: @Eomm

Author: Done
